sdk/deviceprovisioning/azure-iothub-deviceprovisioning/src/main/java/com/azure/deviceprovisioning/implementation/models/Enum1.java

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.deviceprovisioning.implementation.models;
+
+import com.azure.core.util.ExpandableStringEnum;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import java.util.Collection;
+
+/** Defines values for Enum1. */
+public final class Enum1 extends ExpandableStringEnum<Enum1> {
```

Comment:
Code gen gave this enum a very strange name here. @srnagar do you have any insights on why autorest would do this?

Reply:
Here's the relevant part of the swagger, for context
```
{
	"name": "$expand",
	"in": "query",
	"description": "Properties to optionally expand in response.",
	"required": false,
	"type": "string",
	"enum": [
	  "symmetricKey",
	  "x509Certificate",
	  "all"
	]
},
```

Reply:
This could be due to the `$` in the name. Can you add `x-ms-enum` and specify the name you want to use?

`"x-ms-enum": { "name": "EnumName", "modelAsString": true}`

https://github.com/Azure/autorest.testserver/blob/master/swagger/body-array.json#L785

Reply:
That worked, thanks! I'll propose the relevant changes to the service team so they can update their swagger accordingly.